app/settings/data-import/data-import.directive.js

```diff
@@ -1,42 +1,302 @@
 module.exports = [
```

Comment:
This directive is now really hard to read as it has *all* the logic in it. Could you do a pass updating the coding style? ie. use named functions, only bind to scope when needed.

While an external service may not be essential, I think the complicated data munging steps might still be better living elsewhere on a DataImportService or similar. Though just the coding style change might be enough..